### PR TITLE
Added variable definition, arithmetic and assignment operators, and c…

### PR DESCRIPTION
**Pull request by Bufferdash**
Added support for
• Variable definitions 
• Arithmetic operators (+, -, *, /, %, ^) 
• Assignment operators (=, +=, -=, *=, /=, ^=, %=) 
• Conditional statements (<, <=, >, >=, ==, !=, and, or, not)

### CODE REVIEW

File: ANTLR-version/App.g4 [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
 grammar App;
-s : comment   | EOF   ;
-comment : COMMENT NEWLINE s;
+s : comment | definition | NEWLINE s | EOF  ;
+
+comment : COMMENT s;
+definition : ID ASSIGN_OPERATORS equation s ;
+conditional_statement: equation CONDITION equation;
+
+value : ID | NUMBER | '"' ~('"')* '"' ;
```
Comment (YoungBrendanM) [inline ANTLR-version/App.g4]: This should also include booleans ("True" | "False")

File: ANTLR-version/App.g4 [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
+equation : value | value ARITH_OPERATORS equation ;
```
Comment (YoungBrendanM) [inline ANTLR-version/App.g4:10]: alpha * (3 - beta)
Do we need to support parenthesis?

File: ANTLR-version/App.g4 [hunk 3]
```diff
@@ -1,11 +1,23 @@
...
 
 // Lexer Rules
-ARITH_OPERATORS : '+' | '-' | '*' | '/' | '%' | '^' ;
-ASSIGN_OPERATORS : '='  | '+='  | '-=' | '*='   | '/='  | '^=' | '%='   ;
 CONDITION: '<' | '<=' | '>' | '>='  | '==' | '!='    ;
 COMMENT :    '#' ~('\n')*   ;
```
Comment (YoungBrendanM) [inline ANTLR-version/App.g4:14]: add carriage return ('\r") to the negation.

File: ANTLR-version/App.g4 (hunk 1 above)
Comment (YoungBrendanM) [reply ANTLR-version/App.g4]: A conditional statement can technically be set to a value.

**Review by YoungBrendanM — CHANGES_REQUESTED** [review]

File: ANTLR-version/App.g4 (hunk 1 above)
Comment (Bufferdash) [reply ANTLR-version/App.g4]: > A conditional statement can technically be set to a value.

equation supports assigning to a single value.

File: ANTLR-version/App.g4 (hunk 2 above)
Comment (Bufferdash) [reply ANTLR-version/App.g4:10]: > alpha * (3 - beta) Do we need to support parenthesis?

It doesn't require it in the sample file

File: ANTLR-version/App.g4 (hunk 3 above)
Comment (Bufferdash) [reply ANTLR-version/App.g4:14]: For what purpose?

File: ANTLR-version/App.g4 (hunk 1 above)
Comment (Bufferdash) [reply ANTLR-version/App.g4]: > This should also include booleans ("True" | "False")

This is not required in the sample python file

File: ANTLR-version/App.g4 (hunk 3 above)
Comment (YoungBrendanM) [reply ANTLR-version/App.g4:14]: At least on my computer it leaves a \r at the end of every comment

**Review by YoungBrendanM — APPROVED** [review]

**Pull request merged**